src/components/va-text-input/test/va-text-input.e2e.ts

```diff
@@ -0,0 +1,186 @@
+    expect(requiredSpan).not.toBeNull();
+  });
+
+  it('passes an aXe check', async () => {
+    const page = await newE2EPage();
+
+    await page.setContent(
+      '<va-text-input required label="This is a test" error="With an error message"/>',
+    );
+
+    await axeCheck(page);
+  });
+
+  it('fires an analytics event', async () => {
+    const page = await newE2EPage();
+
+    await page.setContent('<va-text-input label="Input Field"/>');
+
+    const analyticsSpy = await page.spyOnEvent('component-library-analytics');
+
+    const inputEl = await page.find('va-text-input >>> input');
+    await inputEl.press('1');
+    await inputEl.press('2');
+    await inputEl.press('3');
+    await inputEl.press('Tab');
+
+    expect(analyticsSpy).toHaveReceivedEventDetail({
+      action: 'blur',
+      componentName: 'va-text-input',
+      details: {
+        label: 'Input Field',
+        value: '123',
+      },
+    });
+  });
+
+  it("doesn't squelch other blur events", async () => {
+    const page = await newE2EPage();
+
+    await page.setContent('<va-text-input label="Input Field"/>');
+
+    const inputEl = await page.find('va-text-input >>> input');
+    const blurSpy = await inputEl.spyOnEvent('blur');
+    await inputEl.press('Tab');
+
+    expect(blurSpy).toHaveReceivedEvent();
+  });
+
+  it("doesn't fire analytics events when disableAnalytics is true", async () => {
+    const page = await newE2EPage();
+
+    await page.setContent(
+      '<va-text-input label="Input Field" disable-analytics />',
+    );
+
+    const analyticsSpy = await page.spyOnEvent('component-library-analytics');
+    const inputEl = await page.find('va-text-input >>> input');
+    await inputEl.press('1');
+    await inputEl.press('Tab');
+
+    expect(analyticsSpy).not.toHaveReceivedEvent();
+  });
+
+  it('adds placeholder text', async () => {
+    const page = await newE2EPage();
+    await page.setContent(
+      '<va-text-input placeholder="Enter your life story" />',
+    );
+
+    // Render the error message text
+    const inputEl = await page.find('va-text-input >>> input');
+    expect(inputEl.getAttribute('placeholder')).toContain(
+      'Enter your life story',
+    );
+  });
+
+  it('adds adds a character limit with descriptive text', async () => {
+    const page = await newE2EPage();
+    await page.setContent('<va-text-input maxlength="3" value="22"/>');
+
+    // Level-setting expectations
+    const inputEl = await page.find('va-text-input >>> input');
+    expect(await inputEl.getProperty('value')).toBe('22');
+    expect(await page.find('va-text-input >>> small')).toBeNull();
+
+    // Test the functionality
+    await inputEl.press('2');
+    expect(await inputEl.getProperty('value')).toBe('222');
+    expect((await page.find('va-text-input >>> small')).innerText).toContain(
+      '(Max. 3 characters)',
+    );
+  });
+
+  it('passes unknown props to the input element in the shadow DOM', async () => {
```

Comment:
The `componentName` thing reminded me to take a gander at some props. We don't have a `name` prop specifically, but that little helper should automatically pass any additional props/attrs onto the `input`. However, _should_ we have a `name` prop? That may be useful when it comes time to get the data out of the form. We can do `myEl.value`, but we won't have anything to match it to.

Reply:
I discovered that `inputEl.name` returned the name, but `vaTextInputEl.name` did not. `vaTextInputEl.getAttribute('name')` returned what we expect, though. :man_shrugging: 

For consistency (and perhaps clearer docs), I just added `name` as a prop. :+1: